lib/galaxy_social.py

```diff
@@ -0,0 +1,181 @@
+            )
+            with open(schema_path, "r") as f:
+                schema = yaml(f)
+            validate(instance=metadata, schema=schema)
+        except Exception as e:
+            raise Exception(f"Invalid metadata in {file_path}.\n{e}")
+
+        metadata["media"] = [media.lower() for media in metadata["media"]]
+
+        for media in metadata["media"]:
+            if not any(
+                item["name"].lower() == media for item in self.plugins_config["plugins"]
+            ):
+                raise Exception(f"Invalid media {media}.")
+
+        metadata["mentions"] = (
+            {key.lower(): value for key, value in metadata["mentions"].items()}
+            if metadata.get("mentions")
+            else {}
+        )
+        metadata["hashtags"] = (
+            {key.lower(): value for key, value in metadata["hashtags"].items()}
+            if metadata.get("hashtags")
+            else {}
+        )
+        markdown_content = markdown(text.strip())
+        plain_content = BeautifulSoup(markdown_content, "html.parser").get_text(
+            separator="\n"
+        )
+        return plain_content, metadata
+
+    def process_markdown_file(self, file_path, processed_files):
+        content, metadata = self.parse_markdown_file(file_path)
+        if self.preview:
+            try:
+                _, _, message = self.plugins["markdown"].create_post(
+                    content,
+                    [],
+                    [],
+                    metadata.get("images", []),
+                    media=metadata["media"],
+                    preview=True,
+                )
+                return processed_files, message
+            except Exception as e:
+                raise Exception(f"Failed to create preview for {file_path}.\n{e}")
+        stats = {}
+        url = {}
+        for media in metadata["media"]:
+            if file_path in processed_files and media in processed_files[file_path]:
+                stats[media] = processed_files[file_path][media]
+                continue
+            mentions = metadata.get("mentions", {}).get(media, [])
+            hashtags = metadata.get("hashtags", {}).get(media, [])
+            images = metadata.get("images", [])
+            stats[media], url[media] = self.plugins[media].create_post(
+                content, mentions, hashtags, images
+            )
+        url_text = "\n".join(
+            [f"[{media}]({link})" for media, link in url.items() if link]
+        )
+        message = f"Posted to:\n\n{url_text}"
+
+        processed_files[file_path] = stats
+        print(f"Processed {file_path}: {stats}")
+        return processed_files, message
+
+    def process_files(self, files_to_process):
+        processed_files = {}
+        message = ""
+        processed_files_path = os.path.join(
+            os.path.dirname(os.path.abspath(__file__)), "..", "processed_files.json"
```

Comment:
better have a --json-out (or similar) param cause if you think of "lib" becoming an installable package at some point, this won't be a good idea.